libs/sdk-ui-charts/src/highcharts/adapter/legendHelpers.ts

```diff
@@ -415,3 +215,81 @@ export function isStackedChart(chartOptions: IChartOptions): boolean {
     const isAreaChartWithOneSerie = isAreaChart(type) && !hasMoreThanOneSeries && !hasStackByAttribute;
     return !isAreaChartWithOneSerie && !isTreemap(type) && Boolean(stacking);
 }
+
+function getLegendDetailsForAutoResponsive(
+    contentRect: ContentRect,
+    legendOptions: ILegendOptions,
+    chartOptions: IChartOptions,
+): ILegendDetails {
+    const width = contentRect?.client?.width;
+    const height = contentRect?.client?.height;
+
+    if (!width || !height) {
+        return null;
+    }
+
+    const name = chartOptions?.legendLabel ? { name: chartOptions?.legendLabel } : {};
+
+    // Decision logic: https://gooddata.invisionapp.com/console/share/KJ2A59MOAQ/548340571
+    if (width < 610) {
```

Comment:
Please name the magic numbers in the function (esp. the 274 - should that be always the same or are those two different values that happen to be the same number?)

Reply:
Added to constants